scipy/sparse/_csr.py

```diff
@@ -223,6 +189,38 @@ def _getcol(self, i):
         return self.__class__((data, indices, indptr), shape=(M, 1),
                               dtype=self.dtype, copy=False)
 
+    def _get_int(self, idx):
+        if 0 <= idx <= self.shape[0]:
+            spot = np.flatnonzero(self.indices == idx)
+            if spot.size:
+                return self.data[spot[0]]
+            return self.data.dtype.type(0)
+        raise IndexError(f'index ({idx}) out of range')
```

Comment:
It's a bit strange to have the error removed from the check so far; it's not necessary to change for this PR, but I think it'd be more intuitive to write:

```
if (idx < 0) or (idx >= self.shape[0]):
    raise IndexError(...)
```

Is the current if check correct? It has `<=` in both positions, but presumably `idx` should be strictly less than `self.shape[0]`.
```

What about negative indices?

Reply:
I agree with your style comment -- and I agree that the `<=` should be `< self.shape[0]`.

Negative indices are handled in the `_validate_indices` method. 

Which reminded me that `_validate_indices` also checks the size of `idx`, so there is no need to check it here. That is why the incorrect `<=` check wasn't flagged by the existing test in `test_indexing1d.py::test_getelement` for index values just beyond the limits of `shape`. 

Thanks for flagging this -- I have removed this "if/raise" statement as it is redundant with logic in `_validate_index`.